### DISCUSSION TITLE
`ruff server` with optional TCP/UNIX socket

### DISCUSSION THREAD

**Discussion by morph027** [astral-sh/ruff#13544] [Ideas]
I'm running `pylsp --tcp` as a service which listens for connections from multiple editor instances instead of creating a new `pylsp` process for every instance which is quite useful.

Switching to the new implementation of `ruff server` instead of `ruff-lsp`, i think, it would be nice to have such a feature too. It obviously needs to handle multiple clients.

**Comment by dhruvmanila** [comment]
What's the use case for this? Is it just to avoid creating a new `ruff server` process for every editor instance? What kind of benefits does that provide? This might be basic questions but I'm just trying to understand the motivation behind this feature request.

**Reply by morph027** [reply]
Jip, basically. In my previous setup, i used one `pylsp` instance, which serves multiple editors e.g. in tilix/tmux/... split windows and therefore saves some resources (not that much i guess).

I could however further improve my editor skills by using splits inside vim/helix which does the same :) So it's not a show stopper at all. Just left it here in case other people think the same.

**Reply by dhruvmanila** [reply]
Are you opening the same project in multiple editor instances? I think that's the only place where this could benefit. If so, I'd recommend using splits inside the editor itself :)

**Comment by bread-mountain4** [comment]
aside from the multiple editors consideration, serving the LSP over either TCP or UDS would help users whose local development is primarily done through docker. so the lsp could be running inside of the container with a local mount for the codebase and the editor could then communicate with the dockerized ruff server via tcp.

**Reply by petterroea** [reply]
Acknowledging that containers are not sufficient as a security boundary, doing all local development inside Docker is a great way of avoiding most modern supply chain attacks with low effort - depending on factors like secret mounting of course. Dev containers have been a thing for a while already and it makes more sense than ever to develop your service out of reach of your personal files and tokens.
